### ISSUE TITLE
Streamlining the `make test`/CI infrastructure?

### ISSUE THREAD

**Issue by jamesmckinna** [agda/agda-stdlib#2248]
Now that we use a merge queueing discipline for PRs, I'm conscious of inefficiencies in the 'mid-/post-review, but pre-merge' phase of commits to PRs, in that the CI seems to do a 'full' cycle not just of typechecking (incl. deprecated modules) and golden testing, but also of all the documentation/HTML index generation, which seems as though it would be more efficient to postpone until *after* adding to the merge queue?

There's a risk of errors only being exposed later in the cycle, but the gain would be faster turnaround in testing pre-merge.

**Comment by MatthewDaggitt** [comment]
Yes, if anyone is willing to take this on to optimise the CI in this way it would be much appreciated!

**Comment by jamesmckinna** [comment]
Currently, I don't have a good mental/actual model of the CI-infrastructure to do this myself. Sorry to be only the generator of 'user' gripes, rather than 'admin' solutions... !

**Comment by jamesmckinna** [comment]
https://github.com/agda/agda/issues/8076
Sigh. Or rather: let's try to prioritise this (somehow)

**Comment by JacquesCarette** [comment]
If this doesn't get resolved soon, I wonder if a hack consisting of moving to an `agda2` organization might fix it?

**Comment by jamesmckinna** [comment]
(UPDATED) Shopping list of 'optimisations':
* absent any automatic validation (eg wrt markdown, or more 'semantically' wrt the rest of the PR), commits solely to `CHANGELOG` (what else? `style-guide`? anything under `doc/`?) shouldn't trigger the full `make test` post-commit
* don't check success for HTML generation until the final addition to the merge queue 
* ditto. possibly also for golden testing (typechecking wrt `README`/`Everything{Safe}` ought to be sufficient for intermediate post-commit testing?)
* [DRY] generate `EverythingSafe` and `EverythingElse`, with `Everything` essentially importing each, as a distinct, moreover simpler, factorisation in both `GenerateEverything` and in the structure of `doc/README.agda`
* [DRY] reconcile the above steps with the generation of `index.agda`
* ... etc.

**Comment by jamesmckinna** [comment]
Are we now finally in a position to close this issue with #3097 , after its various predecessors?

**Comment by silas-hw** [comment]
I would say so!

**Comment by JacquesCarette** [comment]
As per comments above, closing.
